### PR TITLE
Mp 7004/ghost 1click

### PR DESCRIPTION
**Pull request by msannan2**
Migrated Ghost 1-Click over from https://github.com/TryGhost/digitalocean-1-click

##Changes

- Version locked Node Js and Ghost + CLI
- Increase memory requirements
- Updated the template file to include common scripts from the repo and removed the ones bundled with the Ghost fork.
- Minor fixes to the ghost install script
- Added fail2ban and it relevant filters
- Removed yarn since Ghost doesn't need it

### CODE REVIEW

**Review by alexeigutium — APPROVED** [review]

File: ghost-22-04/files/etc/fail2ban/jail.d/ghost.conf [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+port = http,https
+bantime = 1800
+
+[ghost-ratelimit]
+enabled = true
+filter = ghost-ratelimit
+logpath = /var/www/ghost/content/logs/*.log
+maxretry = 3
+port = http,https
+bantime = 86400
```
[the remark this answers is not in the document]
Comment (msannan2) [reply ghost-22-04/files/etc/fail2ban/jail.d/ghost.conf:15]: thanks for pointing it out. I need to uncheck these settings in my IDE

File: common/scripts/010-nodejs.sh [hunk 2]
```diff
@@ -17,4 +17,8 @@ curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
 echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
 
 sudo apt-get -qqy update
-sudo apt-get -qqy install nodejs yarn
+sudo apt-get -qqy install nodejs yarn npm
+
+node -v
+npm -v
+yarn -v
```
[the remark this answers is not in the document]
Comment (msannan2) [reply common/scripts/010-nodejs.sh]: added

File: ghost-22-04/files/etc/fail2ban/filter.d/ghost-auth.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[Definition]
+failregex = ^.*"statusCode":403.*"name":"NoPermissionError".*$
```
[the remark this answers is not in the document]
Comment (msannan2) [reply ghost-22-04/files/etc/fail2ban/filter.d/ghost-auth.conf]: added

File: ghost-22-04/files/etc/fail2ban/filter.d/ghost-ratelimit.conf [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[Definition]
+failregex = ^.*"statusCode":429.*"name":"TooManyRequestsError".*$
```
[the remark this answers is not in the document]
Comment (msannan2) [reply ghost-22-04/files/etc/fail2ban/filter.d/ghost-ratelimit.conf]: added

**Pull request merged**